Shared/SharedUI/BaseDialogs/PackFileTree/PackFileBrowserViewModel.cs

```diff
@@ -29,40 +29,50 @@ public partial class PackFileBrowserViewModel : ObservableObject, IDisposable, I
 
         public ObservableCollection<TreeNode> Files { get; set; } = [];
         public SearchFilter Filter { get; private set; }
-        public ICommand DoubleClickCommand { get; set; }
-        public ICommand ClearTextCommand { get; set; }
 
         [ObservableProperty] TreeNode _selectedItem;
         [ObservableProperty] ObservableCollection<ContextMenuItem2> _contextMenu = [];
 
-        public PackFileBrowserViewModel(IContextMenuBuilder contextMenuBuilder, IPackFileService packFileService, IEventHub? eventHub, bool showCaFiles)
-        {
-            DoubleClickCommand = new RelayCommand<TreeNode>(OnDoubleClick);
-            ClearTextCommand = new RelayCommand(OnClearText);
+        public bool ShowFoldersOnly { get; }
 
+        public PackFileBrowserViewModel(IContextMenuBuilder contextMenuBuilder, IPackFileService packFileService, IEventHub? eventHub, bool showCaFiles, bool showFoldersOnly, bool useEditablePackOnly)
+        {
             _packFileService = packFileService;
             _eventHub = eventHub;
             _contextMenuBuilder = contextMenuBuilder;
 
+            ShowFoldersOnly = showFoldersOnly;
+
             _eventHub?.Register<PackFileContainerSetAsMainEditableEvent>(this, ContainerUpdated);
             _eventHub?.Register<PackFileContainerRemovedEvent>(this, PackFileContainerRemoved);
             _eventHub?.Register<PackFileContainerAddedEvent>(this, x => ReloadTree(x.Container));
             _eventHub?.Register<PackFileContainerFilesUpdatedEvent>(this, Database_PackFilesUpdated);
-            _eventHub?.Register<PackFileContainerFilesAddedEvent>(this, x=> AddFiles(x.Container, x.AddedFiles));
+            _eventHub?.Register<PackFileContainerFilesAddedEvent>(this, x => AddFiles(x.Container, x.AddedFiles));
             _eventHub?.Register<PackFileContainerFilesRemovedEvent>(this, x => Database_PackFilesRemoved(x.Container, x.RemovedFiles));
-                         
+
             _eventHub?.Register<PackFileContainerFolderRemovedEvent>(this, x => Database_PackFileFolderRemoved(x.Container, x.Folder));
             _eventHub?.Register<PackFileContainerFolderRenamedEvent>(this, x => Database_PackFileFolderRenamed(x.Container, x.NewNodePath));
-       
+
             Filter = new SearchFilter(Files);
+            Filter.ShowFoldersOnly = ShowFoldersOnly;
 
             foreach (var item in _packFileService.GetAllPackfileContainers())
             {
                 var loadFile = true;
                 if (!showCaFiles)
                     loadFile = !item.IsCaPackFile;
 
-                if (loadFile)
+                if (useEditablePackOnly)
```

Comment:
this logic hurts my mind, can the if statements be cleaned up to make easier to understnd?  less nesting

Reply:
Yeah that's fair will take another pass at it